### DISCUSSION TITLE
Are modules imported by modules supposed to be transpiled?

### DISCUSSION THREAD

**Discussion by danieltroger** [parcel-bundler/parcel#5807] [Q&A] [answered]
@mischnic just checking in before I open an issue.

I have a module included via file:../../and/so/on, i.e. a symlink in node_modules. That module includes a module from it's node_modules directory. And that module is not being transpiled for my IE11 target, breaking my builds in IE11.

Is that intended behaviour? You said somewhere that modules were not supposed to be transpiled or something in that direction but deleted the comment again. Also they usually are.
There is `"jsnext:main":` in the package.json of the module not being transpiled and maybe that is what creates the issue.

**Comment by mischnic** [comment, marked as answer]
Only source files are transpiled with Babel. Source file means `!fs.realPath(...).includes("node_modules")`

> "jsnext:main"

Only main,module,browser are used

**Comment by danieltroger** [comment]
> Only source files are transpiled with Babel. Source file means !fs.realPath(...).includes("node_modules")

Ah, so the symlink of the first module gets resolved which is why it gets transpiled, but the modules that are actual modules are not. That makes sense.

So it's not an issue. Thanks.

Then I have another question:
How am I supposed to include this polyfill then?
here it contains template strings https://github.com/GoogleChrome/proxy-polyfill/blob/master/src/proxy.js#L103

Is there a way to force transpile it or do I have to move it out of node_modules?

**Reply by mischnic** [reply]
The problem is that some/most packages treat `main` as ES3 in CommonJS and `module` as ES6 (or newer) in ESM. But in reality, it should rather be ES3 in both cases. There isn't really a consensus ([between bundlers](https://jasonformat.com/enabling-modern-js-on-npm/)) so far on how/when node_modules should be transpiled. The actual solution might look something like: packages declare what JS version they are providing in the corresponding fields, and bundlers could then choose to (not) transpile it.

> Is there a way to force transpile it

At the moment: no

See https://github.com/parcel-bundler/parcel/issues/1655

**Reply by danieltroger** [reply]
Shoot, seems like I unknowingly stepped into a very sensitive topic. Thanks for bringing me up to speed!

**Reply by danieltroger** [reply]
Hey @mischnic sorry to bother you again. I can't let go of my dream of transpiled node_modules now. I have more questions.

1. Why can't we transpile node_modules every time?
2. Is [this](https://github.com/parcel-bundler/parcel/issues/1655#issuecomment-775890192) really all that has to be done to get node_modules transpiled? It 100% feels like it is going to have bad side effects
3. How can I get `debugger` statements working in parcel code? I tried running it like this `PARCEL_WORKERS=1 node --inspect --inspect-brk ./parcel` with ./parcel being [this](https://github.com/parcel-bundler/parcel/issues/5773#issuecomment-774167386). But when I add debugger statements to files in `node_modules/@parcel` the debugger doesn't stop there. console.log works so I'm hitting the right files but the debugger doesn't stop. I tried stepping through the code but it also goes through node's internal code and it's impossible to get somewhere. I had `cliclick` run clicking on the "step into" button in a while loop and it hadn't even executed through 5 files after 3 minutes.

**Reply by danieltroger** [reply]
I just cherry picked `PARCEL_WORKERS=1` from https://github.com/parcel-bundler/parcel/blob/v2/CONTRIBUTING.md#notes-and-things-to-be-aware-of, do I need to follow the rest of the guide too? I feel like compiling my own parcel is a bit of a big step

**Reply by mischnic** [reply]
1. There is some discussion here as well: https://twitter.com/AndaristRake/status/1289449348502355968 (and potentially other branches of the original tweet).
2. Yes, I don't see how this could break something *at this time*.
3. You need to do `PARCEL_WORKERS=0` to use no threads whatsoever (I changed that in my open PR).
